src/nlp_expr.jl

```diff
@@ -1097,18 +1111,15 @@ function jump_function(
     ]
 end
 
-# We use `AbstractJuMPScalar` as a catch-all fallback for any mix of JuMP
-# scalars that have not been dispatched by some other method.
-
-function moi_function_type(::Type{<:Vector{<:AbstractJuMPScalar}})
+function moi_function_type(::Type{<:AbstractVector{<:GenericNonlinearExpr}})
     return MOI.VectorNonlinearFunction
 end
 
-function moi_function(f::Vector{<:AbstractJuMPScalar})
+function moi_function(f::AbstractVector{<:GenericNonlinearExpr})
     return MOI.VectorNonlinearFunction(f)
 end
 
-function MOI.VectorNonlinearFunction(f::Vector{<:AbstractJuMPScalar})
+function MOI.VectorNonlinearFunction(f::AbstractVector{<:GenericNonlinearExpr})
```

Comment:
I don't agree. Here, you are explicit about the return type so it should work with a vector of anything

Reply:
But it can be added later in a non breaking way so I won't dispute this

Reply:
Ah true. This is reasonable.